thread/thread.cpp

```diff
@@ -1795,6 +1790,22 @@ R"(
                 return true;
         }
     }
+    int semaphore::signal(uint64_t count) {
+        if (count == 0) return 0;
+        SCOPED_LOCK(splock);
+        auto cnt = m_count.fetch_add(count) + count;
+        if (!q.th) return 0;
+        if (cnt >= q.th->semaphore_count) {
+            do_resume: resume_one();
```

Comment:
`goto do_resume;` here, and move `do_resume: resume_one();` to the end of the function?

Reply:
The following loop depends on the condition that ```q.th``` has been checked and processed. The loop starts from the 2nd element, and uses ```q.th``` as the end.